src/react/src/__tests__/components/DashboardModerationQueueListTable.test.js

```diff
@@ -0,0 +1,175 @@
+    ];
+    
+    const paginatedModerationEvents = [
+        ...sampleModerationEvents,
+        {
+            moderation_id: 13,
+            created_at: '2024-10-15T09:10:20.287Z',
+            name: 'Organic Textiles',
+            country: {
+                name: 'Spain',
+                alpha_2: 'ES',
+                alpha_3: 'ESP',
+                numeric: '724',
+            },
+            contributor_name: 'Textile Innovations',
+            moderation_status: 'RESOLVED',
+            moderation_decision_date: '2024-10-15T09:10:20.287Z',
+            updated_at: '2024-10-15T09:10:20.287Z',
+            source: 'API',
+        },
+        {
+            moderation_id: 14,
+            created_at: '2024-10-16T13:25:40.317Z',
+            name: 'High Tech Components',
+            country: {
+                name: 'Japan',
+                alpha_2: 'JP',
+                alpha_3: 'JPN',
+                numeric: '392',
+            },
+            contributor_name: 'Advanced Manufacturing Co',
+            moderation_status: 'RESOLVED',
+            moderation_decision_date: '2024-10-16T13:25:40.317Z',
+            updated_at: '2024-10-16T13:25:40.317Z',
+            source: 'API',
+        },
+        {
+            moderation_id: 15,
+            created_at: '2024-10-17T07:18:50.327Z',
+            name: 'Bio-Textile Solutions',
+            country: {
+                name: 'Sweden',
+                alpha_2: 'SE',
+                alpha_3: 'SWE',
+                numeric: '752',
+            },
+            contributor_name: 'Eco-Friendly Fabrics',
+            moderation_status: 'RESOLVED',
+            moderation_decision_date: '2024-10-17T07:18:50.327Z',
+            updated_at: '2024-10-17T07:18:50.327Z',
+            source: 'SLC',
+        },
+        {
+            moderation_id: 16,
+            created_at: '2024-10-18T15:45:00.337Z',
+            name: 'CleanTech Machinery',
+            country: {
+                name: 'Australia',
+                alpha_2: 'AU',
+                alpha_3: 'AUS',
+                numeric: '036',
+            },
+            contributor_name: 'Sustainable Machines',
+            moderation_status: 'RESOLVED',
+            moderation_decision_date: '2024-10-18T15:45:00.337Z',
+            updated_at: '2024-10-18T15:45:00.337Z',
+            source: 'API',
+        },
+    ];
+
+    const defaultProps = {
+        events: [],
+        fetching: false,
+    };
+
+    const renderComponent = (props = {}) => renderWithProviders(
+        <Router>
+            <DashboardModerationQueueListTable {...defaultProps} {...props} />,
+        </Router>
+    );
+
+    test('renders loading indicator when fetching', () => {
+        const { getByRole } = renderComponent({ fetching: true });
+
+        expect(getByRole('progressbar')).toBeInTheDocument();
+    });
+
+    test('renders event data in rows', () => {
+        const { getByText } = renderComponent({ events: sampleModerationEvents });
```

Comment:
I'd recommend that for future tests, we set the Redux state using real reducer methods, as demonstrated here: [LoginForm.test.js](https://github.com/opensupplyhub/open-supply-hub/blob/main/src/react/src/__tests__/components/LoginForm.test.js#L75).
This way, you will be able to test the functionality of the reducer methods simply by calling them. Right now, the Redux state is skipped.

Reply:
I used the `renderWithProviders` function, which was wrapped into the `renderComponent`. Moreover, the DashboardModerationQueueListTableHeader is a simple component that doesn't use the Redux state, so I just passed props into it.
You can see it above in the code:
![Screenshot from 2024-11-01 18-02-36](https://github.com/user-attachments/assets/05c2108a-a23a-4de5-b1a9-6954e69a9f57)
What do you think about this solution?

Reply:
Ok, I see.
@mazursasha1990 I just wanted to let you know for future reference that you can use this to test the Redux state.